### ISSUE TITLE
IOS problem with upload

### ISSUE THREAD

**Issue by Smiehoo** [immich-app/immich#9216]
### The bug

On iOS (iPhone 15 Pro) photos are no uploaded on server when I open the app. The upload stuck. I must restart app and then the photos are on timeline but with wrong cloud indicator. Cloud icons show there are only on mobile but I have this photos on server (web). I have second phone with Android 13 and there everything works perfect.

### The OS that Immich Server is running on

Synology DSM

### Version of Immich Server

1.103.1

### Version of Immich Mobile App

1.103.0

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    container_name: immich-srv
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich-mcs
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/hardware-transcoding
    #   file: hwaccel.transcoding.yml 
    #   service: cpu # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    command: [ "start.sh", "microservices" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich-mlr
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    # extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
    #   file: hwaccel.ml.yml
    #   service: cpu # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - /volume1/docker/immich-model:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich-rds
    image: redis:6.2-alpine@sha256:70a7a5b641117670beae0d80658430853896b5ef269ccf00d1827427e3263fa3
    restart: always

  database:
    container_name: immich-db
    image: tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - ${DB_DATA_LOCATION}:/var/lib/postgresql/data
    restart: always
```

### Your .env content

```Shell
x
```

### Reproduction steps

```bash
1. Take a photos.
2. Open app.
3. No new photos are show on timeline but backup icon show thats something is in progress.
4. Detail info about uploading file show no file uploading progress,
5. Restart app.
6. New photos is now on timeline but cloud icon show thats only on mobile (but in fact there is copy on server (visible on web).
7. Logs show error below.
```

### Relevant log output

```shell
PlatformDispatcher - Catch all
type '_Map<String, dynamic>' is not a subtype of type 'String' in type cast
#0 _CastListBase.[] (dart:_internal/cast.dart:99)
#1 ListBase.toList (dart:collection/list.dart:225)
#2 AuditDeletesResponseDto.fromJson (package:openapi/model/audit_deletes_response_dto.dart:54)
#3 ApiClient.fromJson (package:openapi/api_client.dart:262)
#4 ApiClient.deserialize (package:openapi/api_client.dart:158)

#5 AuditApi.getAuditDeletes (package:openapi/api/audit_api.dart:74)

#6 AssetService._getRemoteAssetChanges (package:immich_mobile/shared/services/asset.service.dart:57)

#7 SyncService._syncRemoteAssetChanges (package:immich_mobile/shared/services/sync.service.dart:161)

#8 SyncService.syncRemoteAssetsToDb. (package:immich_mobile/shared/services/sync.service.dart:53)

#9 AssetService.refreshRemoteAssets (package:immich_mobile/shared/services/asset.service.dart:45)

#10 AssetNotifier.getAllAsset (package:immich_mobile/shared/providers/asset.provider.dart:52)
```

**Comment by cgeheim** [comment]
same here also on IOS

**Comment by raisinbear** [comment]
Try to log out in the mobile app and back in. Fixed it for me, also see #9065.

**Comment by schinken** [comment]
Same here on android btw

**Comment by flyotlin** [comment]
same here, but logout/login not fixing the issue

**Comment by domocorn** [comment]
A restart on iOS does not work for me in this instance. iOS 17.2.1 App and server version same as original issue

**Comment by alextran1502** [comment]
@fyfrey What do you think about the error that could cause such parsing issue on `audit_deletes_response_dto`?

**Comment by fyfrey** [comment]
No idea, it's generated open API code. We'd need to see the actual JSON.
The new sync does not use this method/API anymore

**Comment by Smiehoo** [comment]
Login/logout partial works (previous wrong marked photos it is now ok) for couple hours and the problem back.

**Comment by sidamos** [comment]
Just installed Immich and have a similar problem on iOS. I take a new photo, navigate to the app and the photo shows. Then it seems to take forever to automatically upload using the foreground backup.
When I tap on the cloud icon and stop the backup and manually start it again, it is finished super fast.
Workaround for me: While the automatic foreground upload takes forever, lock the screen and unlock it again. Then the upload finishes fast. Seems to have the same effect as stop and start backup.

**Comment by Smiehoo** [comment]
I think problem is not with uploading to server but with detecting new photos on local phone storage and accessing to this new photos. I tested this some more. I turned off autobackup, taked new photos and open Immich app. Immich was not detect that new photos was taken but when I closed app and restart this new photos was showed at once. The manual backup to server was very fast.

**Comment by raisinbear** [comment]
> A login/logout on iOS does not work for me in this instance. iOS 17.2.1 App and server version same as original issue

Right, it took a while, but the issue came back for me, too, on two different iOS devices.

**Comment by alextran1502** [comment]
@raisinbear The fix is in and I believe should fix it in the next release

**Comment by schinken** [comment]
@alextran1502 Just for curiosity do you have a pr/commit link for me?

**Comment by alextran1502** [comment]
@schinken https://github.com/immich-app/immich/issues/9295

**Comment by sidamos** [comment]
Is that fix in version 1.104.0? I have iOS app version 1.104.0 and server version 1.104.0 and the foreground sync is still wonky:
- I take a photo
- I open Immich app
- the photo does not appear but sync starts
- sync takes forever
- I lock and unlock the screen
- a few seconds later sync stops and the photo appears, but it has the wrong icon (cloud without content)
- I force refresh the timeline and then the photo has the right icon (cloud with checkmark)
I can provide logs if needed.

**Comment by alextran1502** [comment]
@sidamos it is fixed and will be available in the next release

**Comment by sidamos** [comment]
@alextran1502 Just checked the release log of 1.104.0 and it says:

> Revert "feat(android) Check server is reachable before starting background backup (https://github.com/immich-app/immich/pull/8989)" by @alextran1502 in https://github.com/immich-app/immich/pull/9324

This is the fix that you mentioned here, so it should be in 1.104.0?

**Comment by alextran1502** [comment]
No, the actual fix is #9479

**Comment by sidamos** [comment]
Still not working reliable with server v.1.105.1 and iOS app v1.105.1. It is a bit better sometimes, but not fixed.
1st try:
- I take a new photo
- I start Immich app
- foreground sync starts
- the new photo appears in the app, cloud icon without checkmark
- sync spins forever
- to fix it, I lock and unlock the phone
- shortly after that, sync is finished
- shortly after that, cloud icon has the checkmark

2nd try:
- I take a new photo
- I start Immich app
- foreground sync starts, 
- new photo does not appear
- sync spins forever
- to fix it, I lock and unlock the phone
- shortly after that, sync is finished
- the new photo appears in the app, cloud icon without checkmark
- after multiple pulls to refresh, the cloud icon has the checkmark

**Comment by Smiehoo** [comment]
> Still not working reliable with server v.1.105.1 and iOS app v1.105.1. It is a bit better sometimes, but not fixed. 1st try:
......"

Yes, it is better, but still, something is not right. I got the same symptoms.

Maybe upload is too soon? Before app, check new photos?

When autobackup is off new photos appear very quick and manual backup is very fast.

**Comment by sidamos** [comment]
Also, background sync works fine (if you have a valid SSL cert).

**Comment by sidamos** [comment]
Since it is not fixed with the latest app version, should we open a new issue?
